PurchasesCoreSwift/Support/ManageSubscriptionModalHelper.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  ManageSubscriptionModalHelper.swift
+//
+//  Created by Andrés Boedo on 16/8/21.
+
+import StoreKit
+
+enum ManageSubscriptionsModalError: Error {
+
+    case couldntGetPurchaserInfo(message: String)
+    case couldntGetWindowScene
+    case storeKitShowManageSubscriptionsFailed(error: Error)
+    case couldntGetAppleSubscriptionsURL
+
+}
+
+extension ManageSubscriptionsModalError: CustomStringConvertible {
+
+    public var description: String {
+        switch self {
+        case .couldntGetPurchaserInfo(let message):
+            return "Failed to get managemementURL from PurchaserInfo. Details: \(message)"
+        case .couldntGetWindowScene:
+            return "Failed to get UIWindowScene"
+        case .storeKitShowManageSubscriptionsFailed(let error):
+            return "error when trying to show manage subscription: \(error.localizedDescription)"
+        case .couldntGetAppleSubscriptionsURL:
+            return "error when trying to form the Apple Subscriptions URL."
+        }
+    }
+
+}
+
+class ManageSubscriptionsModalHelper: NSObject {
+    private let systemInfo: SystemInfo
+    private let purchaserInfoManager: PurchaserInfoManager
+    private let identityManager: IdentityManager
+
+    init(systemInfo: SystemInfo,
+         purchaserInfoManager: PurchaserInfoManager,
+         identityManager: IdentityManager) {
+        self.systemInfo = systemInfo
+        self.purchaserInfoManager = purchaserInfoManager
+        self.identityManager = identityManager
+    }
+
+    @available(iOS 9.0, *)
+    @available(macOS 10.12, *)
+    @available(watchOS, unavailable)
+    @available(tvOS, unavailable)
+    func showManageSubscriptionModal(completion: @escaping (Result<Void, ManageSubscriptionsModalError>) -> Void) {
```

Comment:
this is the main part of the PR: 
check purchaser info. 
if there's already a management url: 
if it's iOS, and iOS 15 is available, open the new native fancy thing
otherwise, open the URL

Reply:
it returns a `Result`, which will either be `void` if successful or an error if anything fails